service/history/workflowTaskHandler.go

```diff
@@ -128,18 +128,61 @@ func newWorkflowTaskHandler(
 func (handler *workflowTaskHandlerImpl) handleCommands(
 	commands []*commandpb.Command,
 ) error {
-
-	for _, command := range commands {
-
+	for _, command := range filterCommandsAfterClose(commands) {
 		err := handler.handleCommand(command)
 		if err != nil || handler.stopProcessing {
 			return err
 		}
 	}
-
 	return nil
 }
 
+func filterCommandsAfterClose(commands []*commandpb.Command) []*commandpb.Command {
```

Comment:
Why server need to actively filter out commands which makes no sense? 

If above is the way to go, then does server also ignore commands like cancel an already fired timer?